aptos-move/block-executor/src/executor.rs

```diff
@@ -116,13 +116,16 @@ where
         let sync_view = LatestView::new(base_view, ViewState::Sync(parallel_state), idx_to_execute);
         let execute_result = executor.execute_transaction(&sync_view, txn, idx_to_execute);
 
-        let mut prev_modified_keys = last_input_output
-            .modified_keys(idx_to_execute)
-            .map_or(HashMap::new(), |keys| keys.collect());
+        let mut prev_modified_keys = match last_input_output.modified_keys(idx_to_execute) {
```

Comment:
why not just map_or_else?

Reply:
issues with type deduction somehow

Reply:
I tried annotating and kept seeing weird errors, will try again

Reply:
worked out, must have been on some weird local state